yt/frontends/gadget/data_structures.py

```diff
@@ -560,8 +620,9 @@ def _is_valid(cls, filename, *args, **kwargs):
         return header.validate()
 
 
-class GadgetHDF5File(ParticleFile):
-
+class GadgetHDF5File(GadgetParticleFile):
+    _file_read_mode = "r"
+    _opening_func = h5py.File
```

Comment:
I am not 100% sure this will work with the on-demand imports.

Reply:
narrator: "it doesn't work."

Reply:
Ok, so I can fix the import error by isolating it with:

```
    @property
    def _opening_func(self):
        return h5py.File
```
but I'll hold off until we finish the discussion on whether the extra `GadgetParticleFile` class simplifies or complicates matters...